### ISSUE TITLE
Using "?" does not cause test to fail

### ISSUE THREAD

**Issue by oknozor** [oknozor/sealed-test#1]
**Reproducer:**

```rust
    #[sealed_test]
    fn test_should_fail() -> Result<(), &'static str> {
        let err = Err("error");
        err?;
        assert!(true);
        Ok(())
    }
```

**Cargo expand:**
```rust
  fn test_should_fail() {
        fn body_fn() -> Result<(), &'static str> {
            let temp_dir = tempfile::TempDir::new().unwrap();
            std::env::set_current_dir(&temp_dir).unwrap();
            let err = Err("error");
            err?;
            if !true {
                ::core::panicking::panic("assertion failed: true")
            };
            Ok(())
        }
        let body: fn() -> Result<(), &'static str> = body_fn;
        fn supervise_fn(child: &mut ::rusty_forkfork::ChildWrapper, _file: &mut ::std::fs::File) {
            ::rusty_forkfork::fork_test::supervise_child(child, 0)
        }
        let supervise: fn(&mut ::rusty_forkfork::ChildWrapper, &mut ::std::fs::File) = supervise_fn;
        ::rusty_forkfork::fork(
            ::rusty_forkfork::fork_test::fix_module_path("sealed_test::tests::test_should_fail"),
            {
                struct _RustyForkId;
                ::rusty_forkfork::RustyForkId::of(::std::any::TypeId::of::<_RustyForkId>())
            },
            ::rusty_forkfork::fork_test::no_configure_child,
            supervise,
            body,
        )
        .expect("forking test failed")
    }
}
```

**Comment by Zshoham** [comment]
Okay, I think I found the problem after looking at the code a little.
When you created rusty-forkfork, you added all the types to handle tests with return values,
but when the fork function calls the test function, it does not look at the output and can't know if it failed.
The point where the test function is called from the forked process is [here](https://github.com/oknozor/rusty-forkfork/blob/77d9506b1af7bd11567e0b0ee2016901667b2a79/src/fork.rs#L117).
`catch_unwind` calls `in_child` (which is the test function) and catches panics that happen in it.
When the code matches on the result of the function, it ignores the data in the `Ok` branch.
If I understand this correctly, all we need to do to fix this is to check in the `Ok` branch if the data returned from the test is an `Err`, and if it is exit with code 70 (same as for a panic).function's return type.

**Comment by Zshoham** [comment]
I created a [pull request](https://github.com/oknozor/rusty-forkfork/pull/1) that should fix this issue, and even work with Option return types as well.

**Comment by oknozor** [comment]
closed by 0f94edbf85f2e13b84e43fe89830951dc62bdd32.
